elide-core/src/test/java/com/yahoo/elide/core/LifeCycleTest.java

```diff
@@ -397,6 +412,80 @@ public void testUpdateWithChangeSpec() {
+        verify(book, never()).onUpdateGenre(any(RequestScope.class), isNull());
+        verify(book, times(1)).preRead(scope);
+        verify(book, never()).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPreSecurityTriggers();
+        verify(book, never()).onCreateBook(scope);
+        verify(book, never()).onDeleteBook(scope);
+        verify(book, times(1)).onUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).preRead(scope);
+        verify(book, never()).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, never()).execute(eq(book), isA(RequestScope.class), eq(Optional.empty()));
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPreCommitTriggers();
+        verify(book, never()).preCreateBook(scope);
+        verify(book, never()).preDeleteBook(scope);
+        verify(book, times(1)).preUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), any());
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.getPermissionExecutor().executeCommitChecks();
+        verify(book, times(2)).checkPermission(scope);
+
+        scope.runQueuedPostCommitTriggers();
+        verify(book, never()).postCreateBook(scope);
+        verify(book, never()).postDeleteBook(scope);
+        verify(book, times(1)).postUpdateGenre(any(RequestScope.class), any(ChangeSpec.class));
+        verify(book, times(1)).postRead(scope);
+        verify(book, times(1)).alwaysOnUpdate();
+        verify(onUpdateDeferredCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdateImmediateCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, times(2)).execute(eq(book), isA(RequestScope.class), any());
+        verify(onUpdatePostCommitCallback, never()).execute(eq(book), isA(RequestScope.class), eq(Optional.empty()));
```

Comment:
Let's do this in more places in the tests.

Reply:
Done